cache-core/src/main/java/io/micronaut/cache/interceptor/CacheInterceptor.java

```diff
@@ -852,13 +857,11 @@ private CompletableFuture<? extends Optional<?>> asyncCacheGet(AsyncCache<?> asy
         if (LOG.isTraceEnabled()) {
             LOG.trace("Getting the value for the key [{}] of the cache [{}]", key, asyncCache.getName());
         }
-        return asyncCache.get(key, requiredType)
-                .exceptionally(throwable ->
-                                       exceptionallyAsync(throwable,
-                                                          () -> errorHandler.handleLoadError(asyncCache,
-                                                                                             key,
-                                                                                             asRuntimeException(throwable)),
-                                                          null));
+        return asyncCache
+            .get(key, requiredType)
+            .exceptionallyAsync(throwable ->
+                exceptionallyAsync(throwable, errorHandler.handleLoadError(asyncCache, key, asRuntimeException(throwable)), Optional.empty())
+            );
```

Comment:
This is the change, we go from having `null` as a default return value to `Optional.empty`.